core/src/main/java/bisq/core/xmr/connection/MoneroConnectionsManager.java

```diff
@@ -0,0 +1,181 @@
+package bisq.core.xmr.connection;
+
+import bisq.core.api.model.UriConnection;
+import bisq.core.btc.setup.WalletConfig;
+import bisq.core.xmr.connection.persistence.model.XmrConnectionList;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+
+
+
+import monero.common.MoneroConnectionManager;
+import monero.common.MoneroRpcConnection;
+
+@Slf4j
+@Singleton
+public final class MoneroConnectionsManager {
+
+    private final Object lock = new Object();
+
+    private final MoneroConnectionManager connectionManager;
+    private final XmrConnectionList connectionList;
+
+    @Inject
+    public MoneroConnectionsManager(MoneroConnectionManager connectionManager,
+                                    XmrConnectionList connectionList) {
+        this.connectionManager = connectionManager;
+        this.connectionList = connectionList;
+        initialize();
+    }
+
+    private void initialize() {
```

Comment:
We should set the connection to the last used connection on initialization.

Reply:
Should we then also store the other configurable states, i.e. `refreshPeriod` and `autoSwitch`?

Reply:
Yes exactly, that would be great.

Allow me to PR my changes so far into your branch, which does not include this change.

Reply:
I opened PRs with cleanup to your branches:

https://github.com/LumnitzF/haveno/pull/1/files
https://github.com/LumnitzF/haveno-ui-poc/pull/1

There are merge conflicts. Guessing the branches just need rebased.

Reply:
Note that the test, "Can manage Monero daemon connections" passes but requires some unreleased fixes to monero-java.